parser/parse.js

```diff
@@ -0,0 +1,182 @@
+
+  let ast = babylon.parse(fileContent, {
+    sourceType: "module",
+    plugins: [
+      "jsx",
+      "objectRestSpread",
+      "classProperties",
+    ]
+  });
+
+  walk.simple(ast, {
+    ImportDeclaration(node) {
+      if (node.source.value === 'react-router' ||
+        node.source.value === 'react-router-native' ||
+        node.source.value === 'react-router-dom') routerExist = true;
+      files.push(node);
+    },
+    JSXIdentifier(node) {
+      components.push(node);
+    },
+    JSXElement(node) {
+      if (node.openingElement.name.name === 'Route' ||
+        node.openingElement.name.name === 'PrivateRoute')
+        routes.push(node.openingElement.attributes);
+    },
+  });
+
+  let temp = filePath.split('/');
+  let currentPath = temp.slice(0, temp.length - 1).join('/') + '/';
+
+  structure[selfID] = new Entity(
+    selfID,
+    temp[temp.length - 1],
+    'file',
+    filePath,
+    []
+  );
+
+  files.forEach(node => {
+    let url = parseFilePath(node.source.value, currentPath);
+    if (!url) return;
+
+    for (let i = 0; i < node.specifiers.length; i++) {
+      componentNames.push(node.specifiers[i].local.name);
+    }
+
+    if (!intel.visited.includes(url)) {
+      let myID = uuid();
+      let temp = url.split('/');
+
+      let name = temp[temp.length - 1] === 'index.js' ?
+        temp.slice(temp.length - 2).join('/') : temp[temp.length - 1];
+      structure[myID] = new Entity(
+        myID,
+        name,
+        'file',
+        url,
+        []
+      );
+      structure[selfID].children.push(myID);
+      parse(url, intel, structure, myID);
+    }
+  });
+
+  components.forEach(node => {
+    if (!componentNames.includes(node.name)) return;
+
+    for (let i = 0; i < structure[selfID].children.length; i++) {
+      if (structure[structure[selfID].children[i]].name === node.name) return;
+    }
+
+    let myID = uuid();
+    structure[myID] = new Entity(
+      myID,
+      node.name,
+      'component',
+      null,
+      []
+    );
+    structure[selfID].children.push(myID);
+  });
+
+  if (routerExist && routes.length && routes.length > 1) {
+    let routerID = uuid();
+    structure[routerID] = new Entity(
+      routerID,
+      'Router',
+      'router',
+      null,
+      []
+    );
+    structure[selfID].children.push(routerID);
+
+    routes.forEach(route => {
+      for (let i = 0; i < route.length; i++) {
+        if (route[i].value &&
+            route[i].value.expression &&
+            componentNames.includes(route[i].value.expression.name)) {
+          let myID = uuid();
+          structure[myID] = new Entity(
+            myID,
+            route[i].value.expression.name,
+            'component',
+            null,
+            []
+          );
+          structure[routerID].children.push(myID);
+          break;
+        }
+      }
+    })
+  }
+
+  return JSON.stringify(structure);
+};
+
+const parseFilePath = (url, currentPath) => {
+  if (!url[0] === '.') return null;
```

Comment:
this syntax is a bit confusing. Are you intending to do url[0] is not equal to '.' or the boolean value of url[0] is equal to '.'?